Sources/Algorithms/Intersperse.swift

```diff
@@ -199,6 +203,195 @@ extension Intersperse: LazySequenceProtocol
 extension Intersperse: LazyCollectionProtocol
   where Base: LazyCollectionProtocol {}
 
+//===----------------------------------------------------------------------===//
+// InterspersedMap
+//===----------------------------------------------------------------------===//
+
+@usableFromInline
+internal struct InterspersedMap<Base: Sequence, Result> {
+  @usableFromInline
+  internal let base: Base
+  
+  @usableFromInline
+  internal let transform: (Base.Element) -> Result
+  
+  @usableFromInline
+  internal let separator: (Base.Element, Base.Element) -> Result
+}
+
+extension InterspersedMap: Sequence {
+  public struct Iterator: IteratorProtocol {
```

Comment:
I assume this `public` access control is accidental?

```suggestion
internal struct Iterator: IteratorProtocol {
```